burn-compute/src/tune/tune_cache.rs

```diff
@@ -20,24 +43,116 @@ pub enum TuneCacheResult<K> {
 
 impl<K: AutotuneKey> TuneCache<K> {
     pub(crate) fn new() -> Self {
-        TuneCache {
-            cache: HashMap::new(),
+        #[cfg(feature = "autotune-persistent-cache")]
+        {
+            let mut cache = TuneCache {
+                cache: HashMap::new(),
+                persistent_cache: HashMap::new(),
+            };
+            if let Err(e) = cache.load() {
+                eprintln!(
+                    "Unable to load autotune cache. Cache will be ignored ({}).",
+                    e
+                );
+            }
+            cache
+        }
+
+        #[cfg(not(feature = "autotune-persistent-cache"))]
+        {
+            TuneCache {
+                cache: HashMap::new(),
+            }
         }
     }
 
     #[allow(clippy::borrowed_box)]
     pub(crate) fn try_cache(
-        &self,
+        &mut self,
         autotune_operation_set: Box<dyn AutotuneOperationSet<K>>,
     ) -> TuneCacheResult<K> {
-        let index = self.cache.get(&autotune_operation_set.key());
-        if let Some(&i) = index {
-            return TuneCacheResult::Hit(autotune_operation_set.fastest(i));
+        let key = autotune_operation_set.key();
+        let result = self.cache.get_mut(&key);
+
+        #[cfg(feature = "autotune-persistent-cache")]
+        {
+            if let Some((is_checked, index)) = result {
```

Comment:
What is actually going on here? I think it would deserve a comment

Reply:
With the last changes I believe the code is now understandable without a comment. Tell me what you think.

Reply:
Yes it's clearer with the struct attribute names.